### DISCUSSION TITLE
JavaScript console errors on dashboard

### DISCUSSION THREAD

**Discussion by grzegorz-uniti** [plausible/analytics#3822] [Self-Hosted Support]
I'm trying to run **v2.1.0-rc.0** of Plausible but it can not render _Dashboard_ because of JavaScript error:
`Uncaught SyntaxError: Unexpected end of input (at app-7a5f917ab82642c117f01bfb42aaea32.js?vsn=d:22:11561)`
generated by _esbuild_ version **0.17.11**.
_Node_ **v21.0.0** / _npm_ **10.4.0**.

![error](https://github.com/plausible/analytics/assets/19939301/2ce6b9e7-d712-4520-9a21-9f843017baf5)

**Comment by ruslandoga** [comment]
👋 @grzegorz-uniti 

What browser are you using? Any extensions? I'm not getting any parsing errors for https://who.edify.space/js/app-7a5f917ab82642c117f01bfb42aaea32.js?vsn=d on Safari. Would you also be able to share the link to your JS file? I'd like to diff them.

**Reply by grzegorz-uniti** [reply]
Chrome 121.0.6167.184 & Safari 17.3.1 - no adblock's/ no ext's.
I think it's bug from _node_module_ or when parsing by _esbuild_.

```
root@analytics:/opt/plausible# mix esbuild default --minify
  ../priv/static/js/dashboard.js      993.2kb
  ../priv/static/js/app.js            148.4kb
  ../priv/static/js/embed.host.js      15.1kb
  ../priv/static/js/embed.content.js   14.3kb
⚡ Done in 169ms
```

My file: 
[dashboard.txt](https://github.com/plausible/analytics/files/14393037/dashboard.txt)

What version of node/ npm you are using?

**Reply by ruslandoga** [reply]
Are you building the scripts yourself?

**Reply by grzegorz-uniti** [reply]
Yes, i don't use Docker. 
I do everything as docker file described.

OS: FreeBSD with Ubuntu environment.

**Reply by ruslandoga** [reply]
The scripts are indeed different: https://gist.github.com/ruslandoga/e9a81d7517333f311a78b08b883c2be6

**Reply by ruslandoga** [reply]
Ok, then please show your full setup, configs, how you compile and set everything up :)

**Reply by ruslandoga** [reply]
> What version of node/ npm you are using?

https://github.com/plausible/analytics/blob/19ce62db617dfafe510f6be72e45fcb37f87d267/.tool-versions#L3

**Reply by grzegorz-uniti** [reply]
**Steps**:
Node v21.0.0 / npm 10.4.0.
1. Install Erlang 26.2.1 with Elixir 1.16.0-otp-26,
2. Install and configure: PostgreSQL & Clickhouse,
3. Download Plausible sources from GitHub,
4. Export config variables into environment,
5. Do `make install` (to create tables in PostgreSQL),
6. Create app directory and do everything like in Dockerfile - step by step to release plausible,
7. After `mix release plausible` a run: `_build/prod/rel/plausible/bin/plausible start`,
8. Running via NGINX proxy.

**NGINX settings**:
```
upstream Analytics {
    server 192.168.0.11:8000;
}

server {
    listen 443 ssl http2;
    server_name XXX;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;
        proxy_set_header X-NginX-Proxy true;

	proxy_pass http://Analytics;
        proxy_redirect off;

        # Socket.IO Support
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
    }
}
```
**Environment variables**:
```
BASE_URL=https://XXX:443
DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/plausible
CLICKHOUSE_DATABASE_URL=http://127.0.0.1:8123/plausible_events_db
SECRET_KEY_BASE=XXX
TOTP_VAULT_KEY=XXX
ENVIRONMENT="prod"
MAILER_ADAPTER=Bamboo.LocalAdapter
LOG_LEVEL=debug
SELFHOST=true
DISABLE_CRON=true
ADMIN_USER_IDS=1
SHOW_CITIES=true
PADDLE_VENDOR_AUTH_CODE=XXX
PADDLE_VENDOR_ID=3942
GOOGLE_CLIENT_ID=XXX
GOOGLE_CLIENT_SECRET=XXX
PROMEX_DISABLED=false
SITE_DEFAULT_INGEST_THRESHOLD=1000000
DISABLE_REGISTRATION=true
```

**XXX** - hidden strings

Everything working except this JavaScript files on Dashboard subpage.

**Reply by grzegorz-uniti** [reply]
I did some tests and it looks like dashboard.js is not provided in its entirety by web server.
Add `/*T*/` to the top and `/*E*/` to the end of `dashboard.js` file, make `mix phx.digest` and file contain this comment's but
when i try to download by web server it has multiple `/*T*/` and no `/*E*/`.

**Reply by ruslandoga** [reply]
Do you download it directly from Plausible or through Nginx?

**Reply by grzegorz-uniti** [reply]
Directly and by proxy - the same behavior.

**Reply by ruslandoga** [reply]
> make mix phx.digest and file contain this comment's

Do you mean the file at `rel/plausible/lib/plausible-0.0.1/priv/static/js/app-7a5f917ab82642c117f01bfb42aaea32.js`? This is the file that would be served when you run it as a release.

**Reply by grzegorz-uniti** [reply]
`/opt/plausible/priv/static/js/dashboard-98391d1d9f81140b7d3cc9c962c80b8f.js`
or
`_build/prod/rel/plausible/lib/plausible-0.0.1/priv/static/js/dashboard-98391d1d9f81140b7d3cc9c962c80b8f.js`
has error when downloading by http server but
`app-XXX.js` is downloading fine.

Maybe is size limit by service?

**Reply by ruslandoga** [reply]
Ah sorry. I got confused since the error in your original post mentioned `app-7a5f917ab82642c117f01bfb42aaea32.js`

**Reply by ruslandoga** [reply]
You need to be checking the files in the release dir, `priv/static` doesn't really matter.

**Reply by grzegorz-uniti** [reply]
Ok, in release dir file `dashboard.js` is fine but when access by http it's broken

**Reply by ruslandoga** [reply]
> Maybe is size limit by service?

AFAIK there are only size limits (8MB) to incoming requests but not outgoing. And the Docker-based installations don't limit the file either, so it's unlikely to be the case.

**Reply by grzegorz-uniti** [reply]
Update _cowboy_ & _plug_cowboy_ with no results :(

**Reply by grzegorz-uniti** [reply]
@ruslandoga i've generate random string file with the same size as `dashboard.js` by `openssl rand -hex 317899 > dashboard-98391d1d9f81140b7d3cc9c962c80b8f.js` and it looks like after 72KB it's crashing.

**Reply by ruslandoga** [reply]
I see. It might be something specific to your setup since `dashboard.js` files are large than 72KB in the released container image and it seems to be working fine:

```console
$ docker compose exec plausible ls -lh lib/plausible-0.0.1/priv/static/js
total 3M
-rw-r--r--    1 plausibl plausibl  148.4K Feb 23 12:36 app-7a5f917ab82642c117f01bfb42aaea32.js
-rw-r--r--    1 plausibl plausibl   47.9K Feb 23 12:36 app-7a5f917ab82642c117f01bfb42aaea32.js.gz
-rw-r--r--    1 plausibl plausibl  148.4K Feb 23 12:36 app.js
-rw-r--r--    1 plausibl plausibl   47.9K Feb 23 12:36 app.js.gz
-rw-r--r--    1 plausibl plausibl  970.2K Feb 23 12:36 dashboard-4f64316823e88f6850f0d0982ecb1899.js
-rw-r--r--    1 plausibl plausibl  310.4K Feb 23 12:36 dashboard-4f64316823e88f6850f0d0982ecb1899.js.gz
-rw-r--r--    1 plausibl plausibl  970.2K Feb 23 12:36 dashboard.js
-rw-r--r--    1 plausibl plausibl  310.4K Feb 23 12:36 dashboard.js.gz
-rw-r--r--    1 plausibl plausibl   14.3K Feb 23 12:36 embed.content-3c876ad2044f50fc31b6be2f75290fd3.js
-rw-r--r--    1 plausibl plausibl    5.4K Feb 23 12:36 embed.content-3c876ad2044f50fc31b6be2f75290fd3.js.gz
-rw-r--r--    1 plausibl plausibl   14.3K Feb 23 12:36 embed.content.js
-rw-r--r--    1 plausibl plausibl    5.4K Feb 23 12:36 embed.content.js.gz
-rw-r--r--    1 plausibl plausibl   15.1K Feb 23 12:36 embed.host-cbfa210be9eb021d87d368ca3773a506.js
-rw-r--r--    1 plausibl plausibl    6.0K Feb 23 12:36 embed.host-cbfa210be9eb021d87d368ca3773a506.js.gz
-rw-r--r--    1 plausibl plausibl   15.1K Feb 23 12:36 embed.host.js
-rw-r--r--    1 plausibl plausibl    6.0K Feb 23 12:36 embed.host.js.gz
```

**Reply by grzegorz-uniti** [reply]
Tried `master` branch, changing version of Erlang and Elixir, change _Cowboy_ settings - no success :(

**Reply by grzegorz-uniti** [reply]
@ruslandoga Ok, i just copied JS files into static web server and change link in template. Temporary fixed.
Can you tell me how to exclude my own IP address from being raported?
As I see in documentation there is an option in settings but i don't have that.
I register new user when everything starts up so I don't know there is an Admin privileged settings?

**Reply by ruslandoga** [reply]
Shields are behind a feature flag right now, but I think you can still access them directly at `/:domain/settings/shields`

**Reply by grzegorz-uniti** [reply]
@ruslandoga yes, it's working fine, thanks!
If anyone have idea how to fix my main issue i will be glad for help. I temporary serving javascript files by another web server.
